client.go

```diff
@@ -123,10 +136,32 @@ func NewClient() *Client {
 	}
 }
 
+// ErrStopRetry is returned by a CheckRetry callback to stop retries, but
+// still have the client return the original error.
+var ErrStopRetry = errors.New("stop retry")
+
 // Do wraps calling an HTTP method with retries.
 func (c *Client) Do(req *Request) (*http.Response, error) {
 	c.Logger.Printf("[DEBUG] %s %s", req.Method, req.URL)
 
+	if c.CheckRetry == nil {
+		// Check the response code. We retry on 500-range responses to allow
```

Comment:
Looks like this comment is duplicated now. Let's reword and squash the two near the spot that actually checks the response code.